controllers/queue_controller.go

```diff
@@ -120,10 +120,16 @@ func (r *QueueReconciler) addFinalizerIfNeeded(ctx context.Context, q *topologyv
 	return nil
 }
 
+// deletes queue from rabbitmq server
+// if server responds with '404' Not Found, it logs and does not requeue on error
+// queues could be deleted manually or gone because of AutoDelete
 func (r *QueueReconciler) deleteQueue(ctx context.Context, client *rabbithole.Client, q *topologyv1alpha1.Queue) error {
 	logger := ctrl.LoggerFrom(ctx)
 
-	if err := validateResponse(client.DeleteQueue(q.Spec.Vhost, q.Spec.Name)); err != nil {
+	err := validateResponseForDeletion(client.DeleteQueue(q.Spec.Vhost, q.Spec.Name))
+	if err == NotFound {
```

Comment:
My preferred method for checking error types in Go is with 
```go
errors.Is(err, NotFound)
```
As it checks the whole call stack of potentially wrapped errors. It doesn't matter for this particular case, since you're only creating one level of errors, so up to you if you want to bother.

Reply:
Done!